libphobos/libdruntime/core/atomic.d

```diff
@@ -1415,6 +1420,11 @@ else version( GNU )
             res = __atomic_compare_exchange_8(cast(void*) here, cast(void*) &ifThis, writeThis,
                                               false, MemoryOrder.seq, MemoryOrder.seq);
         }
+        else static if( T.sizeof == long.sizeof * 2)
+        {
+            res = __atomic_compare_exchange(T.sizeof, cast(void*)here, cast(void*)&ifThis, cast(void*)&writeThis,
+                                            MemoryOrder.seq, MemoryOrder.seq);
```

Comment:
It looks like this needs frontend support to turn the call into `__atomic_compare_exchange_X`, same with `__atomic_load`.